bundles/org.openhab.automation.module.script.graaljs/src/main/java/org/openhab/automation/module/script/graaljs/commonjs/internal/FilesystemFolder.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.module.script.graaljs.commonjs.internal;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * File folder implementation
+ *
+ * @author Jonathan Gilbert - Initial contribution
+ */
+@NonNullByDefault
+public class FilesystemFolder extends AbstractFolder {
```

Comment:
While quickly looking into this PR I noted that the formatting is non-standard here, please format it using Eclipse with the standard formatter.

BTW I'm really happy to see that you got it down to 2200 lines, it makes it reviewable!

Reply:
Apologies that this took me so long, for some reason I didn't see this comment. The code was formatted with the default Intellij settings, but I've reformatted now with the default Eclipse settings (looks like the main difference was 4 spaces vs 2 spaces indentation).